tv_apps/smart-home/test/unit/home_test.js

```diff
@@ -0,0 +1,382 @@
+  var realL10n = navigator.mozL10n;
+  var realMozActivity = window.MozActivity;
+  var searchButton, settingsButton, editButton, timeElem;
+  var filterTabGroup, filterAllButton;
+  var settingsGroup;
+  var subject;
+  var fakeTimer;
+
+  mocksHelperForHomeTest.attachTestHelpers();
+
+  suiteSetup(function() {
+    realL10n = navigator.mozL10n;
+    navigator.mozL10n = MockL10n;
+
+    searchButton = document.createElement('button');
+    searchButton.id = 'search-button';
+    document.body.appendChild(searchButton);
+
+    settingsButton = document.createElement('button');
+    settingsButton.id = 'settings-button';
+    document.body.appendChild(settingsButton);
+
+    editButton = document.createElement('button');
+    editButton.id = 'edit-button';
+    document.body.appendChild(editButton);
+
+    timeElem = document.createElement('div');
+    timeElem.id = 'time';
+    document.body.appendChild(timeElem);
+
+    filterTabGroup = document.createElement('menu-group');
+    filterTabGroup.once = function() {};
+    filterTabGroup.open = function() {};
+    filterTabGroup.close = function() {};
+    filterTabGroup.id = 'filter-tab-group';
+    filterAllButton = document.createElement('smart-button');
+    filterAllButton.id = 'filter-all-button';
+    filterTabGroup.appendChild(filterAllButton);
+    document.body.appendChild(filterTabGroup);
+
+    settingsGroup = document.createElement('settings-group');
+    settingsGroup.once = function() {};
+    settingsGroup.open = function() {};
+    settingsGroup.close = function() {};
+    settingsGroup.id = 'settings-group';
+    document.body.appendChild(settingsGroup);
+  });
+
+  suiteTeardown(function() {
+    navigator.mozL10n = realL10n;
+    document.body.removeChild(searchButton);
+    document.body.removeChild(settingsButton);
+    document.body.removeChild(editButton);
+    document.body.removeChild(timeElem);
+    document.body.removeChild(filterTabGroup);
+
+  });
+
+  setup(function() {
+    subject = new Home();
+    subject.cardScrollable = new XScrollable();
+    subject.folderScrollable = new XScrollable();
+    subject.messageHandler = new MessageHandler();
+    subject.spatialNavigator = new SpatialNavigator();
+    subject.cardManager = new CardManager();
+    subject.searchBar = new SearchBar();
+    subject.filterManager = new FilterManager();
+    subject.edit = new Edit();
+    subject.cardListElem = document.createElement('div');
+    subject.spatialNavigator.m_focusedElement = document.createElement('div');
+    subject.isNavigable = true;
+    subject.settingsButton = settingsButton;
+    subject.editButton = editButton;
+    subject.searchButton = searchButton;
+    subject.settingsGroup = settingsGroup;
+
+    fakeTimer = this.sinon.useFakeTimers();
+    window.MozActivity = MockMozActivity;
+  });
+
+  teardown(function() {
+    window.MozActivity = realMozActivity;
+  });
+
+  suite('visibilityChange > ', function() {
+    teardown(function() {
```

Comment:
I saw you define `visibilityState` for each test under suite `visible >`, do we need to define it again here?